# Canonical has broken their own patch somehow and I don't know how to fix it.
--- message-dialog-restore-traditional-look-on-unity.patch	2014-08-13 15:17:19.738156946 +0900
+++ message-dialog-restore-traditional-look-on-unity.patch	2014-08-13 19:13:04.624876499 +0900
@@ -1,231 +1 @@
-From 9c9626875629093b072f9377b0999ec1407d166c Mon Sep 17 00:00:00 2001
-From: Lars Uebernickel <lars.uebernickel@canonical.com>
-Date: Thu, 29 May 2014 14:52:55 +0200
-Subject: [PATCH 2/2] message dialog: restore traditional look on unity
-
----
- gtk/gtkmessagedialog.c               | 36 ++++++++++++++++++-
- gtk/resources/ui/gtkmessagedialog.ui | 17 +++++++++
- testsuite/a11y/messagedialog.txt     | 67 ++++++++----------------------------
- 3 files changed, 67 insertions(+), 53 deletions(-)
-
-diff --git a/gtk/gtkmessagedialog.c b/gtk/gtkmessagedialog.c
-index e3f8ec4..ca08860 100644
---- a/gtk/gtkmessagedialog.c
-+++ b/gtk/gtkmessagedialog.c
-@@ -104,6 +104,9 @@ struct _GtkMessageDialogPrivate
-   GtkWidget     *label;
-   GtkWidget     *message_area; /* vbox for the primary and secondary labels, and any extra content from the caller */
-   GtkWidget     *secondary_label;
-+  GtkWidget     *icon;
-+  GtkWidget     *box;
-+  GtkWidget     *vbox;
- 
-   guint          has_primary_markup : 1;
-   guint          has_secondary_text : 1;
-@@ -298,6 +301,9 @@ gtk_message_dialog_class_init (GtkMessageDialogClass *class)
-   gtk_widget_class_set_template_from_resource (widget_class, "/org/gtk/libgtk/ui/gtkmessagedialog.ui");
-   gtk_widget_class_bind_template_child_private (widget_class, GtkMessageDialog, label);
-   gtk_widget_class_bind_template_child_private (widget_class, GtkMessageDialog, secondary_label);
-+  gtk_widget_class_bind_template_child_private (widget_class, GtkMessageDialog, image);
-+  gtk_widget_class_bind_template_child_private (widget_class, GtkMessageDialog, box);
-+  gtk_widget_class_bind_template_child_full (widget_class, "dialog-vbox1", FALSE, G_PRIVATE_OFFSET (GtkMessageDialog, vbox));
-   gtk_widget_class_bind_template_child_internal_private (widget_class, GtkMessageDialog, message_area);
- }
- 
-@@ -320,7 +326,27 @@ gtk_message_dialog_init (GtkMessageDialog *dialog)
- G_GNUC_BEGIN_IGNORE_DEPRECATIONS
-   action_area = gtk_dialog_get_action_area (GTK_DIALOG (dialog));
- G_GNUC_END_IGNORE_DEPRECATIONS
--  gtk_button_box_set_layout (GTK_BUTTON_BOX (action_area), GTK_BUTTONBOX_EXPAND);
-+
-+  if (g_strcmp0 (g_getenv ("XDG_CURRENT_DESKTOP"), "Unity") == 0)
-+    {
-+      gtk_container_set_border_width (GTK_CONTAINER (dialog), 6);
-+
-+      gtk_box_set_spacing (GTK_BOX (priv->vbox), 6);
-+
-+      gtk_container_set_border_width (GTK_CONTAINER (priv->box), 0);
-+      gtk_box_set_spacing (GTK_BOX (priv->box), 12);
-+      gtk_widget_set_margin_start (priv->box, 12);
-+      gtk_widget_set_margin_end (priv->box, 12);
-+
-+      gtk_widget_set_halign (priv->label, GTK_ALIGN_START);
-+      gtk_misc_set_alignment (GTK_MISC (priv->label), 0, 0);
-+      gtk_widget_set_margin_top (priv->label, 0);
-+
-+      gtk_widget_set_halign (priv->secondary_label, GTK_ALIGN_START);
-+      gtk_misc_set_alignment (GTK_MISC (priv->secondary_label), 0, 0);
-+    }
-+  else
-+    gtk_button_box_set_layout (GTK_BUTTON_BOX (action_area), GTK_BUTTONBOX_EXPAND);
- }
- 
- static void
-@@ -360,6 +386,7 @@ setup_type (GtkMessageDialog *dialog,
- {
-   GtkMessageDialogPrivate *priv = dialog->priv;
-   const gchar *name = NULL;
-+  const gchar *icon_name = NULL;
-   AtkObject *atk_obj;
- 
-   priv->message_type = type;
-@@ -368,18 +395,22 @@ setup_type (GtkMessageDialog *dialog,
-     {
-     case GTK_MESSAGE_INFO:
-       name = _("Information");
-+      icon_name = "dialog-information-symbolic";
-       break;
- 
-     case GTK_MESSAGE_QUESTION:
-       name = _("Question");
-+      icon_name = "dialog-question-symbolic";
-       break;
- 
-     case GTK_MESSAGE_WARNING:
-       name = _("Warning");
-+      icon_name = "dialog-warning-symbolic";
-       break;
- 
-     case GTK_MESSAGE_ERROR:
-       name = _("Error");
-+      icon_name = "dialog-error-symbolic";
-       break;
- 
-     case GTK_MESSAGE_OTHER:
-@@ -397,6 +428,9 @@ setup_type (GtkMessageDialog *dialog,
-       if (name)
-         atk_object_set_name (atk_obj, name);
-     }
-+
-+  if (icon_name)
-+    gtk_image_set_from_icon_name (GTK_IMAGE (priv->image), icon_name, GTK_ICON_SIZE_DIALOG);
- }
- 
- static void 
-diff --git a/gtk/resources/ui/gtkmessagedialog.ui b/gtk/resources/ui/gtkmessagedialog.ui
-index d75a414..a4b0dda 100644
---- a/gtk/resources/ui/gtkmessagedialog.ui
-+++ b/gtk/resources/ui/gtkmessagedialog.ui
-@@ -30,6 +30,23 @@
-             <property name="margin_end">30</property>
-             <property name="spacing">30</property>
-             <child>
-+              <object class="GtkImage" id="image">
-+                <property name="visible">True</property>
-+                <property name="can_focus">False</property>
-+                <property name="halign">center</property>
-+                <property name="valign">start</property>
-+                <property name="icon_name">image-missing</property>
-+                <property name="use_fallback">True</property>
-+                <property name="icon_size">6</property>
-+                <property name="xpad">12</property>
-+              </object>
-+              <packing>
-+                <property name="expand">False</property>
-+                <property name="fill">False</property>
-+                <property name="position">0</property>
-+              </packing>
-+            </child>
-+            <child>
-               <object class="GtkBox" id="message_area">
-                 <property name="visible">True</property>
-                 <property name="can_focus">False</property>
-diff --git a/testsuite/a11y/messagedialog.txt b/testsuite/a11y/messagedialog.txt
-index a934b40..2366f2d 100644
---- a/testsuite/a11y/messagedialog.txt
-+++ b/testsuite/a11y/messagedialog.txt
-@@ -7,55 +7,6 @@ window1
-   <AtkComponent>
-   layer: window
-   alpha: 1
--  unnamed-GtkContainerAccessible-0
--    "filler"
--    parent: window1
--    index: 1
--    state: enabled horizontal sensitive showing visible
--    toolkit: gtk
--    <AtkComponent>
--    layer: widget
--    alpha: 1
--     
--      "label"
--      parent: unnamed-GtkContainerAccessible-0
--      index: 0
--      name:  
--      state: enabled multi-line sensitive showing visible
--      toolkit: gtk
--      <AtkComponent>
--      layer: widget
--      alpha: 1
--      <AtkText>
--      text:  
--      character count: 1
--      caret offset: 0
--      default attributes: bg-color: <omitted>
--                          bg-full-height: 0
--                          direction: <omitted>
--                          editable: false
--                          family-name: <omitted>
--                          fg-color: <omitted>
--                          indent: 0
--                          invisible: false
--                          justification: left
--                          language: <omitted>
--                          left-margin: 0
--                          pixels-above-lines: 0
--                          pixels-below-lines: 0
--                          pixels-inside-wrap: 0
--                          right-margin: 0
--                          rise: 0
--                          scale: 1
--                          size: <omitted>
--                          stretch: <omitted>
--                          strikethrough: false
--                          style: <omitted>
--                          underline: none
--                          variant: <omitted>
--                          weight: <omitted>
--                          wrap-mode: word
--      <AtkHypertext>
-   dialog-vbox1
-     "filler"
-     parent: window1
-@@ -74,10 +25,23 @@ window1
-       <AtkComponent>
-       layer: widget
-       alpha: 1
-+      image
-+        "icon"
-+        parent: box
-+        index: 0
-+        name: Information
-+        state: enabled sensitive showing visible
-+        toolkit: gtk
-+        <AtkComponent>
-+        layer: widget
-+        alpha: 1
-+        <AtkImage>
-+        image size: 48 x 48
-+        image description: (null)
-       message_area
-         "filler"
-         parent: box
--        index: 0
-+        index: 1
-         state: enabled sensitive showing vertical visible
-         toolkit: gtk
-         <AtkComponent>
-@@ -96,8 +60,7 @@ window1
-           <AtkText>
-           text: Information
-           character count: 11
--          caret offset: 11
--          selection 0: (0, 11) Information
-+          caret offset: 0
-           default attributes: bg-color: <omitted>
-                               bg-full-height: 0
-                               direction: <omitted>
--- 
-2.0.0
-
